controllers/awsmachine_controller.go

```diff
@@ -282,13 +280,6 @@ func (r *AWSMachineReconciler) reconcileNormal(ctx context.Context, machineScope
 		return reconcile.Result{}, nil
 	}
 
-	// TODO(ncdc): move this validation logic into a validating webhook
```

Comment:
I don't see this logic moved anywhere, am i misunderstanding this Webhook?

Reply:
It was in an earlier version of this PR. Not sure why it got deleted?

Reply:
as entire spec is immutable, instead of checking every field of spec, i did this
```
if !reflect.DeepEqual(newAWSMachineTemplate.Spec, oldAWSMachineTemplate.Spec) {
	return errors.New("awsMachineTemplateSpec is immutable")
}
```

Reply:
This validation was being used by the AWSMachine controller, where the current webhook is targeting AWSMachineTemplate.

Reply:
got it!
that means, we need to deploy another validating webhook for `AWSMachine` types, to move these logic, right?

Reply:
correct, thanks :)

Reply:
@tahsinrahman are you going to add the AWSMachine validation webhook in this PR?

Reply:
Sure, I can add it here, or if it becomes too big for review, i can do it in separate pr, which one works best for you? :smile:

Reply:
Either way is fine. If you keep them separate, you can retitle this PR to "Add validating webhook for AWSMachineTemplate"?